### PR TITLE
Adds DNS for new config & adds second BE AppGw

### PR DESCRIPTION
**Pull request by ciaranca**
### Change description ###

Adds DNS for new config & adds second BE AppGw

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: components/cftapps_cluster_lb_backend_arm_tf/dns.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+module "privatedns" {
+  source              = "git::https://github.com/hmcts/azure-private-dns.git//modules/azure-private-dns?ref=master"
+  a_recordsets        = local.a_records
+  env                 = local.dns_zone
+  resource_group_name = "core-infra-intsvc-rg"
+  zone_name           = "service.core-compute-${local.dns_zone}.internal"
+}
+#
```
Comment (timja) [inline components/cftapps_cluster_lb_backend_arm_tf/dns.tf:30]:
```suggestion
```

File: environments/prod/backend_lb_config_arm_to_tf.yaml [hunk 2]
```diff
@@ -9,3 +9,13 @@ gateways:
...
+
+  - gateway_configuration:
+      host_name_suffix: service.core-compute-prod.internal
+      ssl_host_name_suffix: platform.hmcts.net
+      certificate_name: wildcard-platform-hmcts-net
+      private_ip_address: 10.90.96.110
+    app_configuration:
+      # CNP
+      - product: plum
+        component: recipe-backend
```
Comment (timja) [inline environments/prod/backend_lb_config_arm_to_tf.yaml:21]:
```suggestion
    app_configuration: []
```

**Review by timja — COMMENTED** [review]
I think you need to remove plum from here as well:
https://github.com/hmcts/azure-platform-terraform/blob/51884dc2bb788505bfac14481cc84fd9792cc4a9/environments/prod/backend_lb_config.yaml#L310-L311

File: environments/prod/backend_lb_config.yaml [hunk 3]
```diff
@@ -305,10 +305,6 @@ gateways:
       # I&A
       - product: ia
         component: bail-case-api
-      
-      # CNP
-      - product: plum
```
Comment (timja) [inline environments/prod/backend_lb_config.yaml:310]: I would probably remove this in a separate PR so you don't have any race conditions where this PR tries to create the record before it's deleted

**Review by timja — APPROVED** [review]

File: environments/prod/backend_lb_config_arm_to_tf.yaml [hunk 4]
```diff
@@ -9,3 +9,10 @@ gateways:
       # CNP
       - product: plum
         component: recipe-backend
+
```
Comment (ciaranca) [inline environments/prod/backend_lb_config_arm_to_tf.yaml:12]: As no applications added into this second AppGw it was failing plans on the pipeline. SO have commented out until needed.

File: environments/prod/backend_lb_config.yaml (hunk 3 above)
Comment (ciaranca) [reply environments/prod/backend_lb_config.yaml:310]: OK sure, I'll sort that out now, thanks

Comment (ciaranca) [reply environments/prod/backend_lb_config.yaml:310]: removed in - https://github.com/hmcts/azure-platform-terraform/pull/1273

**Review by timja — APPROVED** [review]
this looks like it's going to try create a private dns zone and fail.

```
  # module.privatedns.azurerm_private_dns_zone.zone[0] will be created
  + resource "azurerm_private_dns_zone" "zone" {
      + id                                                    = (known after apply)
      + max_number_of_record_sets                             = (known after apply)
      + max_number_of_virtual_network_links                   = (known after apply)
      + max_number_of_virtual_network_links_with_registration = (known after apply)
      + name                                                  = "service.core-compute-prod.internal"
      + number_of_record_sets                                 = (known after apply)
      + resource_group_name                                   = "core-infra-intsvc-rg"
      + tags                                                  = {
          + "application"  = "core"
          + "builtFrom"    = "hmcts/azure-private-dns"
          + "businessArea" = "CFT"
          + "criticality"  = "High"
          + "environment"  = "production"
        }

      + soa_record {
          + email         = (known after apply)
          + expire_time   = (known after apply)
          + fqdn          = (known after apply)
          + host_name     = (known after apply)
          + minimum_ttl   = (known after apply)
          + refresh_time  = (known after apply)
          + retry_time    = (known after apply)
          + serial_number = (known after apply)
          + tags          = (known after apply)
          + ttl           = (known after apply)
        }
    }
```

I guess fine for this but it will need to be imported, not great though

**Pull request merged**